platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+| Functional Impacts | TSS, Genesis, Cryptography, Reconnect         |
+| Related Proposals  | TSS-Library, TSS-Ledger-Id, TSS-Block-Signing |
+| HIPS               | N/A                                           |
+
+## Purpose and Context
+
+The introduction of the Threshold Signature Scheme (TSS) necessitates a new mechanism for managing node participation in
+consensus and block signing.
+Roster, an immutable subset of the address book, will provide this mechanism, ensuring efficient and secure key
+management for TSS operations.
+This proposal attempts to provide a specification for the behavior of rosters starting from their creation from the
+Candidate Address Book (CAB) to their terminal state within the TSS specification.
+
+The Future Address Book (FAB) is a dynamic list maintained within the App that reflects the desired future state of the
+network's nodes. It will be continuously updated by HAPI transactions, such as those that create, update, or delete
+nodes.
+The CAB is a snapshot of the Future Address Book (FAB), and it is created when the app decides to adopt a new address
+book.
+
+This proposal specifies that Roster(s) will be created from the CAB by the Hedera app and passed to the platform code.
+The mechanism for doing so is detailed below.
+
+### Requirements
+
+- Immutability: Rosters must be immutable to protect the integrity of its computed hash.
+- Efficient Storage: Roster data should be stored efficiently in the state to minimize overhead and unnecessary data
+  copying.
+- Clear API: A well-defined API should be provided for the app to create and submit candidate rosters, and for the
+  platform to manage the lifecycle of submitted rosters.
+- Components update: Platform components that currently rely on the use of the Address book (such as Reconnect, Network,
+  Event Validation etc.) must be adapted to use rosters instead.
+- Address book paradigm discontinued within the platform codebase, only use rosters instead
+
+### Architecture and/or Components
+
+###### Roster API
+
+The Hedera app is already responsible for managing the address book.
+We propose that it continues to do so, and at adoption time, creates a candidate Roster object from the CAB and set it
+in the state.
+The State will contain one map of rosters, keyed by roster’s hash - ```Map<Hash<Roster>, Roster>```.
+It is noteworthy that the roster must be immutable to guarantee the integrity of the computed hash.
+This map of rosters will typically contain the current Active Roster, an optional previously Active Roster, the optional
+previous Candidate Roster (a roster created from a candidate address book), and an optional current Candidate Roster.
+There will be new fields in PlatformState - `candidateRosterHash` and `activeRosterHash` - such that at adoption time,
+the way to trigger the adoption of a roster will be by the client code inserting a roster in the roster map, alongside
+setting the `candidateRosterHash` field in the PlatformState.
```

Comment:
Suppose the app has previously set the `candidateRosterHash` several rounds ago and they set it again several rounds later.  At the time it is set a second time, overriding the first time, how does the platform detect that it has been set again?

Reply:
If it's the exact same one sent before, the hash will exist in the roster map so we will not override it. I've clarified above.